test/gateway/eip20_gateway/progress_stake_with_proof.js

```diff
@@ -0,0 +1,520 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const Gateway = artifacts.require("./TestEIP20Gateway.sol");
+const MockOrganization = artifacts.require('MockOrganization.sol');
+const MockToken = artifacts.require("MockToken");
+const GatewayLib = artifacts.require("GatewayLib");
+
+const BN = require('bn.js');
+const EventDecoder = require('../../test_lib/event_decoder.js');
+const messageBus = require('../../test_lib/message_bus.js');
+const Utils = require('../../../test/test_lib/utils');
+const web3 = require('../../../test/test_lib/web3.js');
+
+const NullAddress = Utils.NULL_ADDRESS;
+const ZeroBytes = Utils.ZERO_BYTES32;
+
+const proofData =  require("../../../test/data/stake_progressed_1.json");
+const MessageStatusEnum = messageBus.MessageStatusEnum;
+
+contract('EIP20Gateway.progressStakeWithProof()', function (accounts) {
+
+  let gateway, mockToken, baseToken, stakeData, progressStakeParams, bountyAmount;
+
+  let setStorageRoot = async function() {
+
+    let blockNumber = new BN(proofData.co_gateway.confirm_stake_intent.proof_data.block_number);
+    let storageRoot = proofData.co_gateway.confirm_stake_intent.proof_data.storageHash;
+    await gateway.setStorageRoot(blockNumber, storageRoot);
+
+    blockNumber = new BN(proofData.co_gateway.progress_mint.proof_data.block_number);
+    storageRoot = proofData.co_gateway.progress_mint.proof_data.storageHash;
+    await gateway.setStorageRoot(blockNumber, storageRoot);
+
+  };
+
+  beforeEach(async function () {
+    mockToken = await MockToken.new({ from: accounts[0] });
+    baseToken = await MockToken.new({ from: accounts[0] });
+
+    let owner = accounts[2];
+    let worker = accounts[7];
+    let organization = await MockOrganization.new(
+      owner,
+      worker,
+      { from: accounts[0] },
+    );
+
+    let coreAddress = accounts[5];
+    let burner = NullAddress;
+
+    bountyAmount = new BN(proofData.gateway.constructor.bounty);
+
+    gateway = await Gateway.new(
+      mockToken.address,
+      baseToken.address,
+      coreAddress,
+      bountyAmount,
+      organization.address,
+      burner,
+    );
+
+    let gatewayLib = await GatewayLib.deployed();
```

Comment:
What's `deployed()`? 🤔

Reply:
Truffle migration deploys the contracts. I am just loading the deployed contract instead of deploying a new one.

Reply:
👍